### ISSUE TITLE
Feature request: a Attachment Infos Function

### ISSUE THREAD

**Issue by macdeport** [RealRaven2000/SmartTemplate4#106]
I'd like to suggest a `%attach_infos()%` function which provides a text like:

    "date-with-xxx-03-03-2021-@-10h20.ics	1.1 KB
     story-file.pdf	78.5 KB"

**Comment by RealRaven2000** [comment]
There is an `%attach( )%` function - which might be a good candidate for extending, can you elaborate about the ics file and the filenames? Is there a significance about "xxx"? What does `%attach_infos()%` do exactly?

**Comment by macdeport** [comment]
@RealRaven2000 I have amended my proposition, hoping to be clearer.

**Comment by RealRaven2000** [comment]
Ok, so you want all attachment listed (if any) with full file name and size in place of the variable...

**Comment by macdeport** [comment]
Exactly 👍

**Comment by RealRaven2000** [comment]
Relevant bug for messages API: https://bugzilla.mozilla.org/show_bug.cgi?id=1550960

If I use this for implementation it will be available only for up-to-date Thunderbird versions (from whenever the new API would land - this could be Tb 78.12 or 87.0b2, not sure yet).

**Comment by macdeport** [comment]
Does the situation is better with Tb 78.14?

**Comment by RealRaven2000** [comment]
> Does the situation is better with Tb 78.14?

no, this is over a year old. we are now in the Thunderbird 91 era. There is an API which by the looks of it was first introduced in 88:

https://webextension-api.thunderbird.net/en/91/messages.html#listattachments-messageid

I think this will be the way forward to reading (and inserting) attachment information

**Comment by macdeport** [comment]
Thank you.
